generate-docs/tools/WhatsNew.ts

```diff
@@ -164,7 +164,7 @@ class APISet {
                     // remove unnecessary parts of the declaration string
                     let newItemText = field.declarationString.replace(/;/g, "");
                     newItemText = newItemText.substring(0, newItemText.lastIndexOf(":")).replace("readonly ", "");
-                    newItemText = newItemText.replace(/\|/g, "\\|");
```

Comment:
Seems like this regex is replacing the first occurrence of a question mark in `newItemText` -- which isn't correct in all cases: 
- it correctly fixes the issue with a question mark being mistakenly appended to a property that's deemed optional in the d.ts 
    - example: [ReplaceCriteria.completeMatch](https://review.docs.microsoft.com/en-us/javascript/api/excel/excel.replacecriteria?branch=master&view=office-js#completematch

        - TypeScript definition: 
![image](https://user-images.githubusercontent.com/3019529/50701963-441a5200-1004-11e9-89a3-329f51a37c02.png)

        - .md table entry: 
![image](https://user-images.githubusercontent.com/3019529/50701935-3369dc00-1004-11e9-91cb-6390ee3fae4b.png)

- but it incorrectly results in the question mark being removed from the first optional parameter in a method

    - example: [RangeAreas.compose(sourceRange, copyType, skipBlanks, transpose)](https://review.docs.microsoft.com/en-us/javascript/api/excel/excel.rangeareas?branch=master&view=office-js#copyfrom-sourcerange--copytype--skipblanks--transpose-)

        - TypeScript definition: 
![image](https://user-images.githubusercontent.com/3019529/50701918-1b925800-1004-11e9-8bf5-5c8e204a60c2.png)

        - .md table entries:  
![image](https://user-images.githubusercontent.com/3019529/50702035-8d6aa180-1004-11e9-9083-c7eb58c6f8e2.png)